jcl/src/java.base/share/classes/openj9/internal/foreign/abi/InternalDowncallHandler.java

```diff
@@ -795,6 +852,8 @@ Object runNativeMethod(Addressable downcallAddr, SegmentAllocator segmtAllocator
 			SetDependency(arena.scope());
 			returnVal = invokeNative(
 					/*[IF JAVA_SPEC_VERSION >= 22]*/
+					heapBases,
+					heapOffsets,
```

Comment:
```java
					(info != null) ? info.bases : null,
					(info != null) ? info.offsets : null,
```

Reply:
Will update all the code around related to `ThreadLocal`.

Reply:
Updated the code with `ThreadLocal` and verified in tests.